pkg/skaffold/build/builder_mux.go

```diff
@@ -59,17 +60,22 @@ func NewBuilderMux(cfg Config, store ArtifactStore, builder func(p latest.Pipeli
 		for _, a := range p.Build.Artifacts {
 			m[a.ImageName] = b
 		}
-		concurrency := b.Concurrency()
-		// set mux concurrency to be the minimum of all builders' concurrency. (concurrency = 0 means unlimited)
-		switch {
-		case minConcurrency < 0:
-			minConcurrency = concurrency
-			logrus.Infof("build concurrency first set to %d parsed from %s[%d]", minConcurrency, reflect.TypeOf(b).String(), i)
-		case concurrency > 0 && (minConcurrency == 0 || concurrency < minConcurrency):
-			minConcurrency = concurrency
-			logrus.Infof("build concurrency updated to %d parsed from %s[%d]", minConcurrency, reflect.TypeOf(b).String(), i)
-		default:
-			logrus.Infof("build concurrency value %d parsed from %s[%d] is ignored since it's not less than previously set value %d", concurrency, reflect.TypeOf(b).String(), i, minConcurrency)
+
+		if cfg.BuildConcurrency() > 0 {
```

Comment:
```suggestion
		if cfg.BuildConcurrency() >= 0 {
```

concurrency can be set to `0` which means run all in parallel.